### PR TITLE
Improved Windows support

### PR DESCRIPTION
**Pull request by ahcorde**
Signed-off-by: ahcorde <ahcorde@gmail.com>

# :factory_worker:  Improvement

## Summary

Improved Windows support. This PR is required here https://github.com/ignitionrobotics/ign-fuel-tools/pull/178

## Checklist
- [x] Signed all commits for DCO
- [x] Added tests
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [x] `codecheck` passed (See [contributing](https://ignitionrobotics.org/docs/all/contributing#contributing-code))
- [ ] All tests passed (See [test coverage](https://ignitionrobotics.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Aignitionrobotics+repo%3Aosrf%2Fsdformat+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge**

### CODE REVIEW

**Comment by ahcorde** [comment]
There is an issue removing a TMP_*.mp4 file in one of tests. This is not making the test to fail but I think we can open an issue. Probably this is because the file is not closed.

File: include/ignition/common/Util.hh [hunk 1]
```diff
@@ -186,6 +186,8 @@ namespace ignition
         void const *_buffer, std::size_t _byteCount);
 
     #ifdef _MSC_VER
+      // Disable warning C4307
```
Comment (mjcarroll) [inline include/ignition/common/Util.hh:189]: Suggestion in https://github.com/ignitionrobotics/ign-common/pull/198

**Review by mjcarroll — CHANGES_REQUESTED** [review]
Can we add a test for the new functionality?

File: include/ignition/common/Util.hh (hunk 1 above)
Comment (mjcarroll) [reply include/ignition/common/Util.hh]: #198 came back clean, so I merged it here.

**Comment by ahcorde** [comment]
> Can we add a test for the new functionality?

@mjcarroll, should we make this function a method (public or private)?

**Review by mjcarroll — APPROVED** [review]

**Comment by mjcarroll** [comment]
> @mjcarroll, should we make this function a method (public or private)?

We talked offline, this doesn't need to be a public method as it is just fixing tests.  Codecov drop is because of Linux vs Windows execution paths.

**Pull request merged**